pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -1156,6 +585,9 @@ def from_sequence(
             evaluation_times,
         )
 
+    def __getattr__(self, name: str) -> Any:
+        return getattr(self.hamiltonian, name)
```

Comment:
I'm not a big fan of how much we are relying on this, it's somewhat hacky. This is giving full access to `Hamiltonian` even though it should be an internal class. Furthermore, it makes public methods not appear in the docs anymore. 
I think some public methods still belong in ` QutipEmulator` (namely those related to the config), For those who don't (eg ` build_operator`), we can make special cases here, for example:
 ```suggestion
    def __getattr__(self, name: str) -> Any:
        if name in exposed_methods:
            return getattr(self.hamiltonian, name)
        else:
            raise AttributeError(...)
```
I know this will probably break a few tests that access private attributes but this is fine imo.

Reply:
Ah I only focused on the issue of keeping backwards compatibility without making too many changes, but I agree it's hacky and I don't feel it is stable ground to keep developing QutipEmulator and Hamiltonian in the future.
I will move the properties and methods llinked with config to QutipEmulator, as well as modify the config type of Hamiltonian to be NoiseModel.
I am thinking of getting rid of the get_attr in QutipEmulator, and repeat all the public methods of Hamiltonian in QutipEmulator - will be perhaps more heavy but surely more explicit